src/btree/bt_cursor.c

```diff
@@ -234,6 +271,17 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint64_t recno, bool *vali
      * update that's been deleted is not a valid key/value pair).
      */
     if (cbt->ins != NULL) {
+        if (WT_CURSOR_HAS_BOUNDS(&cbt->iface) && btree->type == BTREE_ROW) {
+            /* Get the insert list key. */
+            if (key == NULL) {
+                key->data = WT_INSERT_KEY(cbt->ins);
+                key->size = WT_INSERT_KEY_SIZE(cbt->ins);
+            }
+            WT_RET(__key_within_bounds(session, &cbt->iface, key, false, &key_out_of_bounds, NULL));
+            /* The key value pair we were trying to return weren't within the given bounds. */
+            if (key_out_of_bounds)
+                return (0);
+        }
```

Comment:
Set key to NULL 🍔

Reply:
A tmp_key is now used in place of key so this issue goes away.